### DISCUSSION TITLE
How do you iterate over all the root states of a machine ?

### DISCUSSION THREAD

**Discussion by lambda0xff** [statelyai/xstate#2725] [Q&A] [answered]
In my UI, I need an array of all the root states of a given machine.

Right now, I'm doing something like this (in Vue):

```
const { state } = useActor( ...someActor )

const rootStates = Object.keys(state.value.machine!.config.states!)
```

and I'm wondering if there's any more straightforward way to do it ?

**Comment by davidkpiano** [comment, marked as answer]
The `@xstate/graph` package exports a `getStateNodes` function for this:

```js
import { getStateNodes } from '@xstate/graph';

const allStateNodes = getStateNodes(someMachine); 
```
